sklearn/impute/_base.py

```diff
@@ -278,6 +278,10 @@ def _validate_input(self, X, in_fit):
         else:
             dtype = FLOAT_DTYPES
 
+        if not in_fit and self._fit_dtype.kind == "O":
+            # Use object dtype if fitted on object dtypes
+            dtype = self._fit_dtype
```

Comment:
I updated this PR to only use the fit_dtype only if the dype during fit is `object`.

Reply:
This is a good point. Do you think that we should add a unit test regarding the bitness preservation since we try to have it in other cases?

Reply:
I added the test here: [`e97a8df` (#22063)](https://github.com/scikit-learn/scikit-learn/pull/22063/commits/e97a8df67f2d2ece6ace473018eefce110b3fcb3).